### PR TITLE
Switch the Markdown engine from mistune to wenmode

### LINKED ISSUE #323: Evaluate switch from mistune to wenmode
The other day I wanted to check what was the latest version of mistune, and saw the announcement in their README about [wenmode](https://github.com/lepture/wenmode). I tried it in one of my projects, and it's both faster and cleaner to use. As we do use md to html on an "industrial scale" here, it might be worth considering. Also, it has md to md capabilities, and that might be useful for our Markdown output format, but that's another story. First see if it can be a faster but100% backward compatible replacement for mistune.

### PR DESCRIPTION
**Pull request by cboos**
Closes #323.

## What changes

The Markdown engine moves from mistune to [wenmode](https://github.com/lepture/wenmode) 0.15.1 (mistune's successor by the same author). All three pipelines — the two HTML content renderers and the Markdown output's tag protection — share one rule set, `markdown_plugins.transcript_rules()`: wenmode's `github` preset with

- strikethrough restricted to `~~two~~` through `Strikethrough(allow_single_tilde=False)` (transcript prose says "~2, ~6 min" constantly);
- bare-URL autolinks that never link e-mail-shaped tokens (`ruff@0.6.0`, `git@github.com:`);
- wenmode's GFM tag filter off, so every raw-HTML node is fully entity-escaped by the renderer;
- the definition-list plugin, as before.

**SHA linkification** (#156) is one post-parse transform over `Text` and `InlineCode` nodes instead of two order-sensitive inline rules. It links exactly the same SHA multiset as before on a 5936-body corpus.

**The Markdown output no longer re-renders anything.** `linkify_shas_in_text` and `_protect_html_tags` parse with `positions=True` and splice only the byte ranges they change, so the 238-line hand-rolled tokenizer is gone and everything outside the edits is byte-identical to the source. The seven Markdown snapshots are unchanged.

**The escape contract (#245) is kept in full.** Raw HTML is entity-escaped by both HTML renderers. Link targets use a scheme *denylist* (mistune's list plus `search-ms`, `ms-appinstaller`, `ms-msdt`, `intent`, `blob`, `filesystem`, `about`) rather than wenmode's allowlist, which would have dropped editor (`cci:`) and `file:line` targets; a new test pins both directions. Editor schemes such as `vscode:` and `cci:` are a deliberate keep: they need a click and cannot run script in the page. 38 XSS payloads through all three renderers leave no live tag, event handler or unsafe scheme; the browser XSS suite passes unchanged.

## Every rendering difference, classified

This is not bug-for-bug. Old and new pipelines were run over 5936 real message bodies and 957 fixture bodies; every difference is attributed to a cause and labelled in `work/wenmode-evaluation.md` (last section). 143 real and 65 fixture bodies differ, **0 unexplained**, 19 causes:

- **Improvements** (mistune bugs deliberately not reproduced): a bare URL no longer swallows a trailing `**` into its href; backslash-newline is a hard break instead of a literal `\` before `<br />` (Claude Code's shift-enter writes these); `&amp;`/`&copy;` are decoded once instead of shown double-escaped; ragged table rows are no longer dropped; a phantom table from `201 | text` lines that ate a line is gone; list tightness follows CommonMark where mistune rendered loose.
- **Neutral** markup: table cells use `align=` instead of `style=`, list items put a newline before `<p>` and nested blocks, task lists use GFM checkbox markup, indented code keeps its final newline.
- **Regressions: none on wenmode 0.15.1.** On 0.15.0 there were 4 bodies, all wenmode parser bugs affecting spacing or grouping; they are fixed upstream (see below), and on 0.15.1 those 4 bodies match markdown-it-py's structure.

## wenmode 0.15.1

The migration surfaced five wenmode 0.15.0 parser bugs; they were reported on #323 with reproductions and fixed upstream in wenmode 0.15.1, which also added `HTMLRenderer(soft_break="br")` and `Strikethrough(allow_single_tilde=False)`. Pinning `wenmode>=0.15.1,<0.16` removed four local workarounds: a strikethrough subclass, a hard-wrap text handler, a table-rule reordering (and the one table shape it altered), and quote trimming on bare-URL autolinks. Removing them changed no rendered body on either corpus. New tests in `test_markdown_rendering.py` pin each behaviour and fail on 0.15.0's. The remaining differences upstream described as GFM spec behaviour are accepted as such, except the two opt-outs above, which would otherwise strike or link transcript text nobody marked up.

## Tests and snapshots

`just ci` green. Snapshot churn is `+169/-19` with no block added or removed: nine table-cell lines and the footnote-heading CSS block in each page snapshot. `wenmode` is bounded below 0.16 because the extensions subclass its rules and renderer, so a minor bump should be a deliberate, re-tested step. `test_commit_linkifier.py` moves to the wenmode plugin API; one test now asserts CommonMark's tab-indent behaviour instead of pinning the old tokenizer's gap.

## Interaction with #327

The SHA resolver rewrite in #327 changes which SHAs resolve; this PR changes what consumes the answers. On the pair of tips `a91cbc1` + `7787d9a` a combined tree rendered the 5936-body corpus byte-identical with 1080 commit links on both sides, and the linkifier tests from both branches passed together. The commits since then touch neither the resolver calls nor the shared test file, so that result holds by reasoning rather than by re-measurement; re-running it is one command if wanted before the two are merged.

## Not in this PR

Four comments in `git_remote.py` still say "mistune"; they belong with #327, which edits that file. `dev-docs/plugins.md` and `test/_plugins/clmail/README.md` show a `format_html` that returns `None` to fall back to Markdown, which contradicts `HtmlRenderer._dispatch_format` (it synthesizes HTML only when `format_html` is absent); that predates this PR and is a separate doc fix. `scripts/bench_render.py` was not re-run; parser-level speed measured 1.57× and the parser is a small share of render time (#327 is the large one).

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Improvements**
  * Updated server-side Markdown rendering to use the wenmode engine.
  * Improved Markdown handling for HTML escaping, URL safety, syntax highlighting, SHA links, and transcript formatting.
  * Added accessible styling for GitHub-style footnote headings.
* **Bug Fixes**
  * Corrected HTML entity handling and strengthened protection against unsafe URL schemes.
  * Resolved several Markdown parsing and formatting edge cases.
* **Documentation**
  * Updated user, contributor, and developer documentation.
* **Tests**
  * Expanded coverage for Markdown rendering and URL policies.

### CODE REVIEW

File: claude_code_log/html/templates/components/global_styles.css [hunk 1]
```diff
@@ -293,7 +293,22 @@ pre {
...
+ * screen readers; keep it out of the visual flow like GitHub does. */
+.markdown .sr-only,
+.user-md .sr-only {
+    position: absolute;
+    width: 1px;
+    height: 1px;
+    padding: 0;
+    margin: -1px;
+    overflow: hidden;
+    clip: rect(0, 0, 0, 0);
```
[the remark this answers is not in the document]
Comment (cboos) [reply claude_code_log/html/templates/components/global_styles.css]: (Claude) Applied in 8ab5350: `clip-path: inset(50%)` replaces `clip`. The ten page snapshots change by exactly that line each.

File: claude_code_log/markdown_plugins.py [hunk 2]
```diff
@@ -1,440 +1,439 @@
...
+        # Quotes and GFM's own trailing punctuation, in any order
+        # (``…tokenizer",`` ends in a comma *after* the quote).
+        trimmed = value.rstrip("\"'?!.,:*_~")
+        if trimmed == value:
+            return super().parse(parser, text, candidate, state)
+        # Re-match on the shortened text so the base class sees a
+        # candidate whose match ends where the URL does.
+        shorter = self.compiled.match(
+            text[: candidate.start + len(trimmed)], candidate.start
+        )
```
[the remark this answers is not in the document]
Comment (cboos) [reply claude_code_log/markdown_plugins.py:237]: (Claude) Applied in 8ab5350: `wenmode>=0.15.0,<0.16`. The subclasses in `markdown_plugins.py` and the renderer handlers in `html/utils.py` use wenmode internals, so a minor bump should be a deliberate, re-tested step.

File: claude_code_log/html/utils.py [hunk 3]
```diff
@@ -381,97 +388,179 @@ def escape_html(text: str) -> str:
...
+        try:
+            normalized = _normalize_url_for_scheme_check(value)
+            scheme = urlsplit(normalized).scheme.lower()
+        except ValueError:
+            return None
+        if scheme in _HARMFUL_URL_SCHEMES and not normalized.lower().startswith(
+            _GOOD_DATA_PREFIXES
+        ):
+            return None
+        return value
```
[the remark this answers is not in the document]
Comment (cboos) [reply claude_code_log/html/utils.py:504]: (Claude) Declining this one, with a measurement. An allowlist of http, https, irc, ircs, mailto, tel and cci would also drop file-reference links that transcripts carry: `urlsplit('vercel.json:20:8-23:9').scheme` is `'vercel.json'`, and across the 6,893 message bodies used to validate this migration an allowlist would have removed the `href` from 19 link targets (`vercel.json` 8, `next.config.js` 7, `cci` 4). The threat this renderer defends against (#245) is script execution in the page; the denylist covers `javascript:`, `vbscript:`, `data:` (except images) and `file:`, after wenmode's control-character stripping and repeated percent-decoding. It also refuses OS protocol handlers that a click would hand to the desktop (`search-ms`, `ms-appinstaller`, `ms-msdt`, `intent`, `blob`, `filesystem`, `about`), and `test_markdown_url_policy_is_a_denylist` pins both directions. Editor schemes such as `vscode:` are kept deliberately.

**Comment by cboos** [comment]
(Claude) On the two review-summary items:

- **Nitpick, missing `table` rule:** applied in 8ab5350, but as a `RuntimeError` rather than a silent skip. With the new `<0.16` bound it cannot fire today; if a later preset renames the rule, silently losing table rendering would be the worse failure.
- **`dev-docs/plugins.md` / `test/_plugins/clmail/README.md`, `format_html` returning `None`:** correct, the example and the prose contradict `HtmlRenderer._dispatch_format`, which synthesizes HTML only when `format_html` is absent. It is already true on `main`; this PR only renamed the engine in those lines. Left out of scope here and listed as a follow-up in the PR description.

File: dev-docs/plugins.md [hunk 4]
```diff
@@ -233,7 +233,7 @@ class MyToolMessage(ToolUseMessage):
         return f"_(my plugin) action={safe_markdown_inline(action)}_"
 
     def format_html(self, _renderer, _message) -> Optional[str]:
-        return None  # fall back to mistune(format_markdown)
+        return None  # fall back to wenmode(format_markdown)
```
[the remark this answers is not in the document]
Comment (cboos) [reply dev-docs/plugins.md:236]: (Claude) Same finding as in the earlier review summary: correct, and already true on `main`; this PR only renamed the engine in that line. It is listed as a follow-up in the PR description rather than fixed here.

**Comment by cboos** [comment]
(Claude) On the outside-diff item for `html/utils.py` 472-493 (allow only `http`/`https` in `sanitize_url`): declining for the reason given on the earlier thread, where the allowlist finding was withdrawn. Transcript link targets such as `vercel.json:20:8-23:9` parse as custom schemes, and editor schemes are kept on purpose; the denylist rejects script, file, browser-internal and OS-handler schemes, and `test_markdown_url_policy_is_a_denylist` pins both directions. Those lines are unchanged in 09114e5.

**Comment by cboos** [comment]
(Claude) CI note for the record: on 09114e5 the first attempt failed one browser test on Windows 3.12, `test_archive_search_page.py::TestArchiveSearchBrowser::test_search_page_shows_setup_instructions_without_a_server` (Playwright `wait_for_selector('#setup:not([hidden])')` timed out after 10 s); Windows 3.10 and 3.11 were cancelled by fail-fast. The same commit passed that test on Windows 3.13, 3.14 and every Ubuntu job, the test renders no Markdown, and the re-run (attempt 2) is green on all jobs. Treated as a first-sighting timeout; if it recurs, this is where it started.

**Comment by cboos** [comment]
I went through a few large HTML transcripts, and nothing strikes me as broken; quite the opposite- it feels even better (though I couldn't say how or where; it's quite subjective). All good!
